packages/cpp/ArmoniK.Api.Tests/source/SubmitterCLientTest.cpp

```diff
@@ -0,0 +1,225 @@
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "serilog/serilog.h"
+#include "submitter/SessionContext.h"
+#include "submitter/SubmitterClient.h"
+#include "submitter_service.grpc.pb.h"
+#include "utils/GuuId.h"
+#include "utils/StringsUtils.h"
+
+using armonik::api::grpc::v1::TaskOptions;
+using armonik::api::grpc::v1::submitter::CreateSessionReply;
+using armonik::api::grpc::v1::submitter::CreateSessionRequest;
+using armonik::api::grpc::v1::submitter::Submitter;
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+
+using ::testing::_;
+using ::testing::AtLeast;
+
+using namespace armonik::api::common::serilog;
+
+TEST(testMock, createSession) {
+  MockStubInterface stub;
+
+  ClientContext context;
+  CreateSessionReply reply;
+  CreateSessionRequest request;
+
+  const std::vector<std::string>& partition_ids = {"default"};
+
+  TaskOptions task_options;
+
+  std::shared_ptr<Channel> channel = grpc::CreateChannel(
+      "172.30.39.223:5001", grpc::InsecureChannelCredentials());
+
+  task_options.mutable_options()->insert({"key1", "value1"});
+  task_options.mutable_options()->insert({"key2", "value2"});
+  task_options.mutable_max_duration()->set_seconds(3600);
+  task_options.mutable_max_duration()->set_nanos(0);
+  task_options.set_max_retries(3);
+  task_options.set_priority(1);
+  task_options.set_partition_id("cpp");
+  task_options.set_application_name("my-app");
+  task_options.set_application_version("1.0");
+  task_options.set_application_namespace("my-namespace");
+  task_options.set_application_service("my-service");
+  task_options.set_engine_type("Unified");
+
+  ASSERT_EQ(task_options.partition_id(), "cpp");
+
+  EXPECT_CALL(stub, CreateSession(_, _, _)).Times(AtLeast(1));
+
+  SubmitterClient submitter(&stub);
+  std::string session_id =
+      submitter.create_session(task_options, partition_ids);
+
+  std::cout << "create_session response: " << session_id << std::endl;
+
+  ASSERT_TRUE(session_id.empty());
+}
+
+TEST(testMock, cancleSession) {
+  MockStubInterface stub;
+
+  ClientContext context;
+  CreateSessionReply reply;
+  CreateSessionRequest request;
+
+  const std::vector<std::string>& partition_ids = {"default"};
+
+  TaskOptions task_options;
+
+  std::shared_ptr<Channel> channel = grpc::CreateChannel(
+      "172.30.39.223:5001", grpc::InsecureChannelCredentials());
+
+  task_options.mutable_options()->insert({"key1", "value1"});
+  task_options.mutable_options()->insert({"key2", "value2"});
+  task_options.mutable_max_duration()->set_seconds(3600);
+  task_options.mutable_max_duration()->set_nanos(0);
+  task_options.set_max_retries(3);
+  task_options.set_priority(1);
+  task_options.set_partition_id("cpp");
+  task_options.set_application_name("my-app");
+  task_options.set_application_version("1.0");
+  task_options.set_application_namespace("my-namespace");
+  task_options.set_application_service("my-service");
+  task_options.set_engine_type("Unified");
```

Comment:
This is duplicated. You could most likely create a function (or static method) to create such a task options.